lib/vrl/compiler/src/shared_value.rs

```diff
@@ -0,0 +1,62 @@
+use std::cell::{Ref, RefCell, RefMut};
+use std::rc::Rc;
+
+use crate::Value;
+
+mod convert;
+mod path;
+mod target;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct SharedValue(pub(crate) Rc<RefCell<Value>>);
```

Comment:
`SharedValue` is just a wrapper over `Rc<RefCell<Value>> `. 

Since it is used everywhere now it is much easier to create a wrapper and then add functionality to the wrapper than it is to use `Rc<RefCell<Value>>` everywhere and then have to deal with the `Rc<RefCell` directly.